### ISSUE TITLE
tf.nn.avg_pool3d crashes(floating point exception) when `input` contain large value and stride=0

### ISSUE THREAD

**Issue by DNXie** [tensorflow/tensorflow#42205]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below):2.1.0
- Python version:3.6.9
- Bazel version (if compiling from source):N/A
- GCC/Compiler version (if compiling from source):N/A
- CUDA/cuDNN version:N/A
- GPU model and memory:N/A

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with:
1. TF 1.0: `python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"`
2. TF 2.0: `python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**

`tf.nn.avg_pool3d` crashes(floating point exception) when `input` contain large value and `stride=0`. Related #42206

**Describe the expected behavior**

Expect no crashes

**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.
~~~python
import tensorflow as tf
import numpy as np
input = tf.constant( [[[[[1e+40]]]]], dtype=np.float64)
tf.nn.avg_pool3d(input=input,ksize=1,strides=0,padding='SAME')
~~~
**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.
~~~python
Floating point exception (core dumped)
~~~
Related #42206

**Comment by DNXie** [comment]
Check the [gist](https://colab.research.google.com/drive/1RP1JkN2nuGmLI1_dsAqZSrA8SajEzjhO#scrollTo=SBuqgS_T0IG7)

**Comment by Saduf2019** [comment]
@DNXie 
Please provide access to the link shared.

**Comment by DNXie** [comment]
@Saduf2019  Done. And you can also reproduce it with the code I provided in the post.

**Comment by Saduf2019** [comment]
I am able to reproduce this issue, please find the [gist here](https://colab.research.google.com/gist/Saduf2019/64733016a242ed3e261b72b57e0f8347/untitled367.ipynb)

**Comment by ymodak** [comment]
This is fixed with latest tf-nightly. It requires that strides to pooling ops be non-zero and that its length match the tensor rank.
See the [gist](https://colab.research.google.com/gist/ymodak/b4c7e1bab69b573628740d3b244fecbc/untitled367.ipynb)
Thanks!
